examples/tv-casting-app/darwin/MatterTvCastingBridge/MatterTvCastingBridge/CastingServerBridge.mm

```diff
@@ -261,7 +277,7 @@ - (void)getDiscoveredCommissioner:(int)index
 {
     ChipLogProgress(AppServer, "CastingServerBridge().getDiscoveredCommissioner() called");
 
-    dispatch_async(_chipWorkQueue, ^{
+    dispatch_sync(_chipWorkQueue, ^{
```

Comment:
why do we change some of these to sync calls? Is there something dangerous about using the async?

should we add a comment here so future readers can understand the change?

Reply:
I alluded to the reasons for this in the PR description. The reason I made this API make the response callbacks synchronously is to avoid breaking clients if they write their code like this:
```
var commissioner: DiscoveredNodeData?;
            repeat {
                castingServerBridge.getDiscoveredCommissioner(i, clientQueue: DispatchQueue.main, discoveredCommissionerHandler: { (result: DiscoveredNodeData?) -> () in
                    commissioner = result;
                     // .. do something with commissioner
                    }
                })
                i += 1
            } while(commissioner != nil)
```

While we can require clients to handle async callbacks, it felt unnecessary in the case of this API because the results are immediately available from the cache on the SDK (no network calls are required). If this makes sense, I can add that this API will make callbacks to the client code synchronously in the API docs.

Reply:
See https://github.com/project-chip/connectedhomeip/pull/24046/commits/1b8dcc16ea1800e40dd7b0d2a82548dfdf731bb1